guides/v2.1/coding-standards/docblock-standard-general.md

```diff
@@ -326,27 +328,29 @@ class Profiler
 ### Functions and methods
 {:#functions-methods}
 
-Functions and methods must have:
+In general, typed method signature should be preferred over PHPDoc annotations when possible.
 
-* Short description
-* Long description that explains the motivation behind the implementation.
+Functions and methods should have:
+
+* A short description in case it adds meaningful information beyond the method name.
+* If the purpose of the method is not obvious, a long description that explains the motivation behind the implementation.
```

Comment:
Some sentence to involve "describe why method is implemented and not how", please